src/actors/satellite-detector/satellite_detector.py

```diff
@@ -0,0 +1,211 @@
+
+EXAMPLES = '''
+- name: Detect satellite installation
+  satellite_detector: ""
+  register: detection_result
+'''
+
+RETURN = '''
+satellite_detected:
+    description: A flag determining whether a satellite installation was detecgted
+    type: boolean
+
+satellite_services:
+    description: An array holding lost of services if satellite was found. This field is not
+                 not provided, if there is no satellite detected.
+    type: array of strings
+
+satellite_version_major:
+    description: An major version of found satellite instance. This field is not
+                 not provided, if there is no satellite detected.
+    type: string
+
+satellite_version_minor:
+    description: An minor version of found satellite instance. This field is not
+                 not provided, if there is no satellite detected.
+    type: string
+
+satellite_version_bugfix:
+    description: An bugfix version of found satellite instance. This field is not
+                 not provided, if there is no satellite detected.
+    type: string
+'''
+
+
+class DetectionFailed(RuntimeError):
+    pass
+
+
+def extract_version(version_string, result):
+    version_sep = version_string.split(".")
+    result["satellite_version_major"] = version_sep[0]
+    result["satellite_version_minor"] = version_sep[1]
+    result["satellite_version_bugfix"] = ".".join(version_sep[2:])
+
+
+def execute_cmd(cmd, shell=False):
+    if not shell:
+        cmd = shlex.split(cmd)
+
+    process = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=shell)
+    out, err = process.communicate()
+
+    return (process, out, err)
+
+
+def detect_satellite5():
+    verif_process, verif_out, _ = execute_cmd("rpm -q satellite-schema --qf %{VERSION}")
+
+    if not verif_process.returncode:
+        result = dict(
+            satellite_detected=True,
+        )
+
+        # Version
+        extract_version(verif_out, result)
+
+        # Services
+        services_process, serv_out, _ = execute_cmd(
+            cmd='eval \"$('
+                'sed -e \"s/exit/:/\" -e \"s/\$0/\$PWD)/g\" $('
+                'which rhn-satellite))\" 2>&1 > /dev/null; echo -n $SERVICES',
+            shell=True)
+
+        if services_process.returncode:
+            raise DetectionFailed()
+
+        if serv_out is not None:
+            result["satellite_services"] = serv_out.split(" ")
+
+            # Optional services
+            services_process, _, _ = execute_cmd("chkconfig --list dhcpd | grep on", shell=True)
+            if not services_process.returncode:
+                result["satellite_services"].append("dhcp")
+
+            services_process, _, _ = execute_cmd("chkconfig --list --type xinetd tftp | grep on", shell=True)
+            if not services_process.returncode:
+                result["satellite_services"].append("tftp")
+
+            return result
+
+    return None
+
+
+def detect_satellite6():
+    verif_process, verif_out, _ = execute_cmd("rpm -q satellite --qf %{VERSION}")
```

Comment:
Small note: info about installed packages, versions, ... should be provided in future by some common actor. In addition, it should be checked, whether it is really rpm signed by RH or whether it is just some rpm with same name.  If you want to be independent now on that info, it's ok, just suggest to remember it or write something like `#TODO input rpm` comment above those lines you expect it will be possible to get as input in future, so it will be easy in future to find out such cases and replace them. Analogically for `chkconfig`.